src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1140,26 +1130,48 @@ bool OrbitMainWindow::eventFilter(QObject* watched, QEvent* event) {
   return QObject::eventFilter(watched, event);
 }
 
-void OrbitMainWindow::closeEvent(QCloseEvent* event) {
+bool OrbitMainWindow::ShowCloseWarningAndReturnResult() {
+  return QMessageBox::question(
+             this, QApplication::applicationName(),
+             "This discards any unsaved progress. Are you sure you want to continue?") ==
+         QMessageBox::Yes;
+}
+
+bool OrbitMainWindow::ShowCaptureAbortWarningAndReturnResult() {
+  return QMessageBox::question(this, "Capture in progress",
+                               "A capture is currently in progress. Do you want to abort the "
+                               "capture and exit Orbit?") == QMessageBox::Yes;
+}
```

Comment:
Nit: Not sure if these two functions make it more readable to me. I tend to say no, since they contain a single statement.

Reply:
I agree and inlined them